server/chat-plugins/chatlog.ts

```diff
@@ -180,20 +180,25 @@ export const LogViewer = new class {
 		return this.linkify(buf);
 	}
 
-	renderDayResults(results: {[day: string]: SearchMatch[]}) {
-		const renderResult = (match: SearchMatch) => (
-			this.renderLine(match[0]) +
-			this.renderLine(match[1]) +
-			`<div class="chat chatmessage highlighted">${this.renderLine(match[2])}</div>` +
-			this.renderLine(match[3]) +
-			this.renderLine(match[4])
-		);
+	renderDayResults(results: {[day: string]: SearchMatch[]}, roomid: RoomID) {
+		const renderResult = (match: SearchMatch) => {
+			if (!match[2]) return null;
```

Comment:
This should probably be rejected before being considered a `SearchMatch` in the first place.